### ISSUE TITLE
Allow `await` in global (module) scope

### ISSUE THREAD

**Issue by NiklasGollenstede** [mozilla/addons-linter#3741]
### Describe the problem and steps to reproduce it:

Putting an `await` in a global module scope causes a syntax error, failing the validation (and thus submission on AMO).
[proposal-top-level-await](https://github.com/tc39/proposal-top-level-await) was actually just updated to Stage 4, and works in Firefox 89 (which is to be released any day now) and Chrome.
So as far as mainline Firefox is concerned, this *is* valid.

I guess one could use this as a test file `background.js`:
```js
import foo from 'foo';
await null;
export const bar = 42;
```

### What happened?

Critical validation error:
```
JavaScript syntax error
(Parsing as module error: Cannot use keyword 'await' outside an async function at ...)
(Parsing as script error: 'import' and 'export' may appear only with 'sourceType: module' at ...)
```

### What did you expect to happen?

The JS file should be parsable.

### Anything else we should know?

I had actually been holding back on porting my AMD modules to ES modules precisely because of the lack of an `await` before `export`ing. Now I ported my code, and can't upload it.

**Comment by Rob--W** [comment]
We're relying on `eslint` to parse the code. Since top-level-await has been promoted to stage 4, I expect eslint to follow shortly:

- https://github.com/eslint/eslint/issues/13178
- https://github.com/eslint/eslint/discussions/13934

**Comment by NiklasGollenstede** [comment]
Well, that was quick, I assume you had this on your radar already :+1: 

And true, to use "fancy" syntax with `eslint`, you have to use a different parser, like `babel` or `tsc`.
We'll just have to wait ...

**Comment by Rob--W** [comment]
Depends on https://github.com/eslint/eslint/issues/14632

**Comment by ioanarusiczki** [comment]
@willdurand 
I tested with the addon attached:
I'm seeing the JavaScript syntax error present on AMO stage. 

![stage](https://user-images.githubusercontent.com/33448286/140509943-4683f4d5-d7c6-436d-b1f0-735a6172829d.png)

On AMO dev or linter 4.0.0 I don't see any errors being present:

![dev](https://user-images.githubusercontent.com/33448286/140509981-b72df08d-0122-4a61-b203-c40df8305c66.png)

![linter 4 0 0](https://user-images.githubusercontent.com/33448286/140509999-30f5cbf8-b308-4f41-a78e-17c7e149364a.png)



[background.zip](https://github.com/mozilla/addons-linter/files/7485941/background.zip)
